providers/base/bin/wifi_nmcli_test.py

```diff
@@ -351,29 +407,49 @@ def print_journal_entries(start):
 
     args = parser.parse_args()
 
-    start_time = datetime.datetime.now()
+    return args
 
-    cleanup_nm_connections()
+
+def main():
+    args = parser_args()
+    start_time = datetime.datetime.now()
     device_rescan()
-    count = list_aps(args)
+    if hasattr(args, "essid"):
+        print("Scanning for APs with ESSID: {}".format(args.essid))
+        aps_dict = list_aps(args.device, args.essid)
+    else:
+        print("Scanning for all APs")
+        aps_dict = list_aps(args.device)
+    show_aps(aps_dict)
 
     if args.test_type == "scan":
-        if count == 0:
+        if not aps_dict:
             print("Failed to find any APs")
-            sys.exit(1)
+            return 1
         else:
-            print("Found {} access points".format(count))
-            sys.exit(0)
+            print("Found {} access points".format(len(aps_dict)))
+            return 0
+
+    if not aps_dict:
+        print("Targed access points: {} found".format(args.essid))
```

Comment:
are you going to say `targeted access points not found` here?

Reply:
ohhh, your are right~~
Thanks for noticing this~~